api/src/main/java/org/eclipse/microprofile/faulttolerance/CircuitBreaker.java

```diff
@@ -69,15 +78,29 @@
 public @interface CircuitBreaker {
 
     /**
-     * Defines the failure criteria.
+     * The list of exception types which should be considered failures
      * <p>
-     * A method call will be considered a failure if it throws an exception and the type of that exception is assignable
-     * to any of the types listed in {@code failOn}.
+     * Note that if a method throws a {@link Throwable} which is not a subclass of either {@link Error} or {@link Exception}, non-portable behavior
```

Comment:
I'm stealing this sentence for the `@Retry` javadoc :-) and I found that it relies on people understanding that subclassing is reflexive. In the spec text, I mention explicitly that `Throwable` itself is custom throwable, while `Error` and `Exception` are not. Here, I don't want to get unnecessarily verbose, and I want the text to be accessible. Perhaps we can say

```
Note that if a method throws a {@link Throwable} which is not an {@link Error} or {@link Exception}, non-portable behavior results.
```

or

```
Note that if a method throws a {@link Throwable} which is not an instance of {@link Error} or {@link Exception}, non-portable behavior results.
```

(In the second case, relying on people knowing how the `instanceof` operator behaves.)

WDYT?

Reply:
That's a good point.

Another possibility I've seen in other specs would be
```
Note that if a method throws a {@link Throwable} which is not assignable to either {@link Error} or {@link Exception}, non-portable behavior results.
```

On balance, I think I prefer using your first suggestion in the javadoc as I find it easier to understand, even if the "assignable to" language is more precise.

Reply:
I'd be fine with using the "assignable to" variant as well.

Reply:
Updated this to match #446

Reply:
Perhaps GitHub is flaky, but it only shows me that it's updated on 1 place out of 3.

Reply:
err, yeah, sorry about that, should be fixed now.